transport/transport.go

```diff
@@ -330,12 +330,20 @@ func NewServerTransport(protocol string, conn net.Conn, maxStreams uint32, authI
 	return newHTTP2Server(conn, maxStreams, authInfo)
 }
 
+// DefaultDialerFn is the default dialer.
+func DefaultDialerFn(addr string, timeout time.Duration, cancel <-chan struct{}) (net.Conn, error) {
```

Comment:
you can write these functions in the tests instead of exporting them. Transport is only used by grpc package. internal package is just used to host some test only methods.